src/main/java/bio/terra/service/filedata/azure/AzureSynapsePdao.java

```diff
@@ -152,15 +167,38 @@
         Statement statement = connection.createStatement();
         ResultSet resultSet = statement.executeQuery(query)) {
       var refIds = new ArrayList<String>();
-      while (resultSet.next()) {
-        refIds.add(resultSet.getString(refColumn.getName()));
+      switch (collectionType) {
+        case SNAPSHOT:
+          while (resultSet.next()) {
+            extractFileIdFromDrs(resultSet.getString(refColumn.getName())).ifPresent(refIds::add);
+          }
+          break;
+        case DATASET:
+          while (resultSet.next()) {
+            String refId = resultSet.getString(refColumn.getName());
+            if (!StringUtils.isEmpty(refId)) {
+              refIds.add(refId);
+            }
+          }
+          break;
       }
       return refIds;
     } catch (SQLException ex) {
       throw new AzureResourceException("Could not query dataset table for fileref columns", ex);
     }
   }
 
+  private Optional<String> extractFileIdFromDrs(final String drsUri) {
+    if (StringUtils.isEmpty(drsUri)) {
```

Comment:
Oh, I agree but I didn't think we wanted to use syntax like that

Reply:
Here it seems fine to me since the method is returning `Optional`. The case I'd rather avoid is where the code doesn't need to return `Optional` but is using it to avoid `if (x != null)`.